daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilInputContentHandler.scala

```diff
@@ -0,0 +1,250 @@
+  private val characterData = new StringBuilder()
+  private var prefixMapping: NamespaceBinding = _
+
+  def getUnparseResult: DFDL.UnparseResult = unparseResult
+
+  def enableInputterUriAbsolutization(): Unit = inputter.enableUriAbsolutization = true
+
+  override def setDocumentLocator(locator: Locator): Unit = {
+    // do nothing
+  }
+
+  override def startDocument(): Unit = {
+    infosetEvent.eventType = One(StartDocument)
+    sendToInputter()
+  }
+
+  override def endDocument(): Unit = {
+    infosetEvent.eventType = One(EndDocument)
+    sendToInputter()
+  }
+
+  override def startPrefixMapping(prefix: String, uri: String): Unit = {
+    val pre = if (prefix == "") null else prefix
+    prefixMapping = NamespaceBinding(pre, uri, prefixMapping)
+  }
+
+  override def endPrefixMapping(prefix: String): Unit = {
+    // do nothing
+  }
+
+  override def startElement(uri: String, localName: String, qName: String, atts: Attributes): Unit = {
+    // we need to check if the characters data is all whitespace, if it is we drop the whitespace
+    // data, if it is not, it is an error as starting a new element with actual characterData means
+    // we haven't hit an endElement yet, which means we're in a complexElement and a complexElement
+    // cannot have character content
+    if (characterData.nonEmpty && characterData.toString().trim.nonEmpty) {
+      throw new IllegalContentWhereEventExpected("Non-whitespace characters in complex " +
+        "Element: " + characterData.toString()
+      )
+    } else {
+      // reset since it was whitespace only
+      characterData.setLength(0)
+    }
+
+    if (!infosetEvent.isEmpty && infosetEvent.localName.isDefined) {
+      // we started another element while we were in the process of building a startElement
+      // this means the first element was complex and we are ready for the inputter queue
+      sendToInputter()
+    }
+    // use Attributes to determine xsi:nil value
+    val nilIn = atts.getIndex(XMLConstants.W3C_XML_SCHEMA_INSTANCE_NS_URI, "nil")
+    infosetEvent.nilValue = if (nilIn >= 0) {
+      val nilValue = atts.getValue(nilIn)
+      One(nilValue)
+    } else {
+      Nope
+    }
+    // set localName and namespaceURI
+    val qNameArr = qName.split(":")
+    infosetEvent.localName =
+      if (localName.nonEmpty) {
+        One(localName)
+      } else if (qNameArr.length > 1) {
+        qNameArr.lift(1)
+      } else if (qNameArr.nonEmpty) {
+        qNameArr.lift(0)
+      } else {
+        Nope
+      }
+    infosetEvent.namespaceURI =
+      if (uri.nonEmpty) {
+        One(uri)
+      } else if (qNameArr.length > 1) { // has a prefix
+        // get the prefix off the qname
+        val qNamePrefix = qNameArr.lift(0).get
+        // look up prefix with and without xmlns prefix
+        val in = atts.getIndex(s"xmlns:$qNamePrefix")
+        val otherIn = if (in == -1) atts.getIndex("qNamePrefix") else in
```

Comment:
I think this is looking for the XML attribute on the element named ``xmlns:foo="..."`` to figure out the namespace. Though, it isn't clear to me why this is needed.. Shouldn't ``startPrefixMapping`` be called for each of the xmlns mappings in the infoset, so our namepace bindings are all available there? So this could be somethign like this?
```scala
prefixMapping.getURI(qNamePrefix)
```

Reply:
Ah good catch. That was a typo!

@stevedlawrence , the XMLReader implementation we use doesn't set the namespaces feature to be true by default, so we have no uris/localNames, and start/endPrefixMappings don't get called and we must depend instead on the AttributesImpl object. So this implementation is for the use case where the user's XMLReader implementation also has non-standard feature implementation. So AttributeImpl usually looks like "xmlns:prefix", so the prefix is stored with the namespaceURI, and we can look it up this way. 

We do check if the prefixMapping has the qNamePrefix though, we just check if after we check the atts.

```scala
name = "localpart="xmlns:ex",rawname="xmlns:ex""
type = "CDATA"
value = "http://example.com"
nonNormalizedValue = "http://example.com"
```

Reply:
Gotcha, it would be worth adding comment explaining why we have to do all this logic.